File: push/src/main/java/com/xiaomi/channel/commonutils/string/XMStringUtils.java
package com.xiaomi.channel.commonutils.string;

import android.text.TextUtils;
import java.io.UnsupportedEncodingException;
import java.math.BigInteger;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.Collection;
import java.util.Iterator;
import java.util.List;
import java.util.Random;
import java.util.regex.Pattern;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/channel/commonutils/string/XMStringUtils.class */
public class XMStringUtils {
    private static final char MAX_ASCII = 127;
    private static final char MIN_ASCII = 0;

    public static String bytesToString(byte[] bArr) {
        if (bArr == null || bArr.length <= 0) {
            return null;
        }
        try {
            return new String(bArr, "UTF-8");
        } catch (UnsupportedEncodingException e) {
            return new String(bArr);
        }
    }

    public static boolean checkAllAscii(String str) {
        if (str == null) {
            return true;
        }
        for (int i = 0; i < str.length(); i++) {
            char cCharAt = str.charAt(i);
            if (cCharAt < 0 || cCharAt > MAX_ASCII) {
                return false;
            }
        }
        return true;
    }

    public static boolean contains(String str, String str2) {
        int i = 0;
        int i2 = 0;
        while (i < str2.length() && i2 < str.length()) {
            if (str2.charAt(i) == str.charAt(i2)) {
                i++;
                i2++;
            } else {
                i2++;
            }
        }
        return i == str2.length();
    }

    public static String generateRandomString(int i) {
        Random random = new Random();
        StringBuffer stringBuffer = new StringBuffer();
        for (int i2 = 0; i2 < i; i2++) {
            stringBuffer.append("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789".charAt(random.nextInt("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789".length())));
        }
        return stringBuffer.toString();
    }

    public static byte[] getBytes(String str) {
        try {
            return str.getBytes("UTF-8");
        } catch (UnsupportedEncodingException e) {
            return str.getBytes();
        }
    }

    public static String getHexString(byte[] bArr) {
        String str = "";
        for (byte b : bArr) {
            str = str + Integer.toString((b & 255) + 256, 16).substring(1);
        }
        return str;
    }

    public static String getMd5(byte[] bArr) {
        String str;
        try {
            MessageDigest messageDigest = MessageDigest.getInstance("MD5");
            messageDigest.update(bArr);
            str = String.format("%1$032X", new BigInteger(1, messageDigest.digest()));
        } catch (Exception e) {
            str = "";
        }
        return str.toLowerCase();
    }

    public static String getMd5Digest(String str) {
        if (TextUtils.isEmpty(str)) {
            return "";
        }
        try {
            MessageDigest messageDigest = MessageDigest.getInstance("MD5");
            messageDigest.update(getBytes(str));
            return String.format("%1$032X", new BigInteger(1, messageDigest.digest()));
        } catch (NoSuchAlgorithmException e) {
            return str;
        }
    }

    public static String getSHA1Digest(String str) {
        if (str == null) {
            return null;
        }
        try {
            MessageDigest messageDigest = MessageDigest.getInstance("SHA1");
            messageDigest.update(getBytes(str));
            return String.format("%1$032X", new BigInteger(1, messageDigest.digest()));
        } catch (NoSuchAlgorithmException e) {
            return str;
        }
    }

    public static String getStringNotNull(String str) {
        if (TextUtils.isEmpty(str)) {
            str = "";
        }
        return str;
    }

    public static int getStringUTF8Length(String str) {
        if (TextUtils.isEmpty(str)) {
            return 0;
        }
        try {
            return str.getBytes("UTF-8").length;
        } catch (UnsupportedEncodingException e) {
            return 0;
        }
    }

    public static boolean isNumberAndLetter(String str) {
        if (TextUtils.isEmpty(str)) {
            return false;
        }
        return Pattern.compile("^[A-Za-z0-9]+$").matcher(str).matches();
    }

    public static boolean isTheSameChars(String str) {
        if (TextUtils.isEmpty(str)) {
            return false;
        }
        char cCharAt = str.charAt(0);
        for (int i = 1; i < str.length(); i++) {
            if (str.charAt(i) != cCharAt) {
                return false;
            }
        }
        return true;
    }

    public static String join(Collection<?> collection, char c) {
        if (collection == null) {
            return null;
        }
        return join(collection.iterator(), c);
    }

    public static String join(Collection<?> collection, String str) {
        if (collection == null) {
            return null;
        }
        return join(collection.iterator(), str);
    }

    public static String join(Iterator<?> it, char c) {
        if (it == null) {
            return null;
        }
        if (!it.hasNext()) {
            return "";
        }
        Object next = it.next();
        if (!it.hasNext()) {
            return next.toString();
        }
        StringBuffer stringBuffer = new StringBuffer(256);
        if (next != null) {
            stringBuffer.append(next);
        }
        while (it.hasNext()) {
            stringBuffer.append(c);
            Object next2 = it.next();
            if (next2 != null) {
                stringBuffer.append(next2);
            }
        }
        return stringBuffer.toString();
    }

    public static String join(Iterator<?> it, String str) {
        if (it == null) {
            return null;
        }
        if (!it.hasNext()) {
            return "";
        }
        Object next = it.next();
        if (!it.hasNext()) {
            return next.toString();
        }
        StringBuffer stringBuffer = new StringBuffer(256);
        if (next != null) {
            stringBuffer.append(next);
        }
        while (it.hasNext()) {
            if (str != null) {
                stringBuffer.append(str);
            }
            Object next2 = it.next();
            if (next2 != null) {
                stringBuffer.append(next2);
            }
        }
        return stringBuffer.toString();
    }

    public static String join(Object[] objArr) {
        return join(objArr, (String) null);
    }

    public static String join(Object[] objArr, char c) {
        if (objArr == null) {
            return null;
        }
        return join(objArr, c, 0, objArr.length);
    }

    public static String join(Object[] objArr, char c, int i, int i2) {
        if (objArr == null) {
            return null;
        }
        int i3 = i2 - i;
        if (i3 <= 0) {
            return "";
        }
        StringBuffer stringBuffer = new StringBuffer(i3 * ((objArr[i] == null ? 16 : objArr[i].toString().length()) + 1));
        for (int i4 = i; i4 < i2; i4++) {
            if (i4 > i) {
                stringBuffer.append(c);
            }
            if (objArr[i4] != null) {
                stringBuffer.append(objArr[i4]);
            }
        }
        return stringBuffer.toString();
    }

    public static String join(Object[] objArr, String str) {
        if (objArr == null) {
            return null;
        }
        return join(objArr, str, 0, objArr.length);
    }

    public static String join(Object[] objArr, String str, int i, int i2) {
        if (objArr == null) {
            return null;
        }
        String str2 = str;
        if (str == null) {
            str2 = "";
        }
        int i3 = i2 - i;
        if (i3 <= 0) {
            return "";
        }
        StringBuffer stringBuffer = new StringBuffer(i3 * ((objArr[i] == null ? 16 : objArr[i].toString().length()) + str2.length()));
        for (int i4 = i; i4 < i2; i4++) {
            if (i4 > i) {
                stringBuffer.append(str2);
            }
            if (objArr[i4] != null) {
                stringBuffer.append(objArr[i4]);
            }
        }
        return stringBuffer.toString();
    }

    public static String obfuscateString(String str, int i) {
        if (TextUtils.isEmpty(str)) {
            return "";
        }
        int length = str.length();
        int i2 = i;
        if (i <= 0 || length < i) {
            i2 = length / 3;
            if (i2 <= 1) {
                i2 = 1;
            } else if (i2 > 3) {
                i2 = 3;
            }
        }
        StringBuilder sb = new StringBuilder();
        for (int i3 = 0; i3 < length; i3++) {
            if ((i3 + 1) % i2 == 0) {
                sb.append('*');
            } else {
                sb.append(str.charAt(i3));
            }
        }
        return sb.toString();
    }

    public static int stringToInt(String str, int i) {
        try {
            return Integer.parseInt(str);
        } catch (NumberFormatException e) {
            return i;
        }
    }

    public static int[] toIntArray(List<Integer> list) {
        int[] iArr = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            iArr[i] = list.get(i).intValue();
        }
        return iArr;
    }

    public static long[] toLongArray(List<Long> list) {
        long[] jArr = new long[list.size()];
        for (int i = 0; i < list.size(); i++) {
            jArr[i] = list.get(i).longValue();
        }
        return jArr;
    }

    public static String[] toStrArray(List<String> list) {
        String[] strArr = new String[list.size()];
        list.toArray(strArr);
        return strArr;
    }

    public static String toUpperCase(String str) {
        return TextUtils.isEmpty(str) ? "" : str.toUpperCase();
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/channel/commonutils/string/XMStringUtils.kt
package com.xiaomi.channel.commonutils.string

import android.text.TextUtils
import java.io.UnsupportedEncodingException
import java.math.BigInteger
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException
import java.util.*
import java.util.regex.Pattern

object XMStringUtils {
    private const val MAX_ASCII: Char = 127.toChar()
    private const val MIN_ASCII: Char = 0.toChar()
    private const val ALPHANUMERIC = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"

    @JvmStatic
    fun bytesToString(bArr: ByteArray?): String? {
        if (bArr == null || bArr.isEmpty()) return null
        return try {
            String(bArr, charset("UTF-8"))
        } catch (e: UnsupportedEncodingException) {
            String(bArr)
        }
    }

    @JvmStatic
    fun checkAllAscii(str: String?): Boolean {
        if (str == null) return true
        for (i in str.indices) {
            val c = str[i]
            if (c < MIN_ASCII || c > MAX_ASCII) {
                return false
            }
        }
        return true
    }

    @JvmStatic
    fun contains(str: String, str2: String): Boolean {
        var i = 0
        var i2 = 0
        while (i < str2.length && i2 < str.length) {
            if (str2[i] == str[i2]) {
                i++
                i2++
            } else {
                i2++
            }
        }
        return i == str2.length
    }

    @JvmStatic
    fun generateRandomString(length: Int): String {
        val random = Random()
        val sb = StringBuilder(length)
        repeat(length) {
            sb.append(ALPHANUMERIC[random.nextInt(ALPHANUMERIC.length)])
        }
        return sb.toString()
    }

    @JvmStatic
    fun getBytes(str: String): ByteArray {
        return try {
            str.toByteArray(charset("UTF-8"))
        } catch (e: UnsupportedEncodingException) {
            str.toByteArray()
        }
    }

    @JvmStatic
    fun getHexString(bArr: ByteArray): String {
        var str = ""
        for (b in bArr) {
            str += Integer.toString((b.toInt() and 255) + 256, 16).substring(1)
        }
        return str
    }

    @JvmStatic
    fun getMd5(bArr: ByteArray): String {
        return try {
            val messageDigest = MessageDigest.getInstance("MD5")
            messageDigest.update(bArr)
            String.format("%1\$032X", BigInteger(1, messageDigest.digest())).lowercase()
        } catch (e: Exception) {
            ""
        }
    }

    @JvmStatic
    fun getMd5Digest(str: String): String {
        if (TextUtils.isEmpty(str)) return ""
        return try {
            val messageDigest = MessageDigest.getInstance("MD5")
            messageDigest.update(getBytes(str))
            String.format("%1\$032X", BigInteger(1, messageDigest.digest()))
        } catch (e: NoSuchAlgorithmException) {
            str
        }
    }

    @JvmStatic
    fun getSHA1Digest(str: String?): String? {
        if (str == null) return null
        return try {
            val messageDigest = MessageDigest.getInstance("SHA1")
            messageDigest.update(getBytes(str))
            String.format("%1\$032X", BigInteger(1, messageDigest.digest()))
        } catch (e: NoSuchAlgorithmException) {
            str
        }
    }

    @JvmStatic
    fun getStringNotNull(str: String?): String {
        return if (TextUtils.isEmpty(str)) "" else str!!
    }

    @JvmStatic
    fun getStringUTF8Length(str: String?): Int {
        if (TextUtils.isEmpty(str)) return 0
        return try {
            str!!.toByteArray(charset("UTF-8")).size
        } catch (e: UnsupportedEncodingException) {
            0
        }
    }

    @JvmStatic
    fun isNumberAndLetter(str: String?): Boolean {
        if (TextUtils.isEmpty(str)) return false
        return Pattern.compile("^[A-Za-z0-9]+$").matcher(str!!).matches()
    }

    @JvmStatic
    fun isTheSameChars(str: String?): Boolean {
        if (TextUtils.isEmpty(str)) return false
        val firstChar = str!![0]
        for (i in 1 until str.length) {
            if (str[i] != firstChar) {
                return false
            }
        }
        return true
    }

    @JvmStatic
    fun join(collection: Collection<*>?, delimiter: Char): String? {
        if (collection == null) return null
        return join(collection.iterator(), delimiter)
    }

    @JvmStatic
    fun join(collection: Collection<*>?, delimiter: String): String? {
        if (collection == null) return null
        return join(collection.iterator(), delimiter)
    }

    @JvmStatic
    fun join(it: Iterator<*>?, c: Char): String? {
        if (it == null) return null
        if (!it.hasNext()) return ""
        val next = it.next()
        if (!it.hasNext()) return next.toString()
        val sb = StringBuffer(256)
        if (next != null) sb.append(next)
        while (it.hasNext()) {
            sb.append(c)
            val next2 = it.next()
            if (next2 != null) sb.append(next2)
        }
        return sb.toString()
    }

    @JvmStatic
    fun join(it: Iterator<*>?, str: String): String? {
        if (it == null) return null
        if (!it.hasNext()) return ""
        val next = it.next()
        if (!it.hasNext()) return next.toString()
        val sb = StringBuffer(256)
        if (next != null) sb.append(next)
        while (it.hasNext()) {
            if (str != null) sb.append(str)
            val next2 = it.next()
            if (next2 != null) sb.append(next2)
        }
        return sb.toString()
    }

    @JvmStatic
    fun join(objArr: Array<Any?>): String? {
        return join(objArr, null as String?)
    }

    @JvmStatic
    fun join(objArr: Array<Any?>, c: Char): String? {
        if (objArr == null) return null
        return join(objArr, c, 0, objArr.size)
    }

    @JvmStatic
    fun join(objArr: Array<Any?>, c: Char, start: Int, end: Int): String? {
        if (objArr == null) return null
        val count = end - start
        if (count <= 0) return ""
        val sb = StringBuffer(count * ((if (objArr[start] == null) 16 else objArr[start].toString().length) + 1))
        for (i4 in start until end) {
            if (i4 > start) sb.append(c)
            if (objArr[i4] != null) sb.append(objArr[i4])
        }
        return sb.toString()
    }

    @JvmStatic
    fun join(objArr: Array<Any?>, str: String?): String? {
        if (objArr == null) return null
        return join(objArr, str, 0, objArr.size)
    }

    @JvmStatic
    fun join(objArr: Array<Any?>, str: String?, start: Int, end: Int): String? {
        if (objArr == null) return null
        val delimiter = str ?: ""
        val count = end - start
        if (count <= 0) return ""
        val sb = StringBuffer(count * ((if (objArr[start] == null) 16 else objArr[start].toString().length) + delimiter.length))
        for (i4 in start until end) {
            if (i4 > start) sb.append(delimiter)
            if (objArr[i4] != null) sb.append(objArr[i4])
        }
        return sb.toString()
    }

    @JvmStatic
    fun obfuscateString(str: String?, i: Int): String {
        if (TextUtils.isEmpty(str)) return ""
        val length = str!!.length
        var i2 = i
        if (i <= 0 || length < i) {
            i2 = length / 3
            if (i2 <= 1) {
                i2 = 1
            } else if (i2 > 3) {
                i2 = 3
            }
        }
        val sb = StringBuilder()
        for (i3 in 0 until length) {
            if ((i3 + 1) % i2 == 0) {
                sb.append('*')
            } else {
                sb.append(str[i3])
            }
        }
        return sb.toString()
    }

    @JvmStatic
    fun stringToInt(str: String?, default: Int): Int {
        return try {
            str?.toInt() ?: default
        } catch (e: NumberFormatException) {
            default
        }
    }

    @JvmStatic
    fun toIntArray(list: List<Int>): IntArray {
        return list.map { it.toInt() }.toIntArray()
    }

    @JvmStatic
    fun toLongArray(list: List<Long>): LongArray {
        return list.map { it.toLong() }.toLongArray()
    }

    @JvmStatic
    fun toStrArray(list: List<String>): Array<String> {
        return list.toTypedArray()
    }

    @JvmStatic
    fun toUpperCase(str: String?): String {
        return if (TextUtils.isEmpty(str)) "" else str!!.uppercase()
    }
}
